components/tikv_util/src/thread_name_prefix.rs

```diff
@@ -0,0 +1,150 @@
+//! Centralized definitions of TiKV thread name prefixes.
+//!
+//! TiKV spawns many worker threads across subsystems (raftstore, scheduler,
+//! CDC, backup, PD workers, etc.) via `std::thread::Builder`. Historically
+//! these names were scattered as ad-hoc string literals, making it harder to:
+//!   - audit all TiKV-owned threads,
+//!   - keep naming consistent,
+//!   - update names globally,
+//!   - recognize threads easily in tools like `top`, GDB/LLDB, `perf`, etc.,
+//!   - learn the DB internals for developers who are just getting started.
+//!
+//! This file gathers all TiKV-controlled thread name **prefixes** in a single
+//! place. Components should reference these constants when spawning threads.
+//!
+//! Threads created by external libraries (e.g., RocksDB, gRPC, Prometheus) are
+//! excluded since TiKV cannot rename them.
+//!
+//! Constants are ordered by when TiKV creates these threads during server
+//! startup.
+
+pub const ARCHIVE_WORKER_THREAD: &str = "archive-worker";
+
+pub const SLOGGER_THREAD: &str = "slogger";
+
+pub const TIME_MONITOR_THREAD: &str = "time-monitor";
+
+pub const GRPC_SERVER_THREAD: &str = "grpc-server";
+
+pub const PD_MONITOR_THREAD: &str = "pdmonitor";
+
+pub const TSO_WORKER_THREAD: &str = "tso-worker";
+
+pub const TIMER_THREAD: &str = "timer";
+
+pub const BACKTRACE_LOADER_THREAD: &str = "backtrace-loader";
+
+pub const BACKGROUND_WORKER_THREAD: &str = "background";
+
+pub const CHECK_LEADER_THREAD: &str = "check-leader";
+
+pub const REGION_COLLECTOR_WORKER_THREAD: &str = "region-collector-worker";
+
+pub const SST_RECOVERY_THREAD: &str = "sst-recovery";
+
+pub const FLOW_CHECKER_THREAD: &str = "flow-checker";
+
+pub const GC_WORKER_THREAD: &str = "gc-worker";
+
+pub const CDC_THREAD: &str = "cdc";
+
+pub const PD_WORKER_THREAD: &str = "pd-worker";
+
+pub const UNIFIED_READ_POOL_THREAD: &str = "unified-read-pool";
+
+pub const DEBUGGER_THREAD: &str = "debugger";
+
+pub const RESOURCE_METERING_RECORDER_THREAD: &str = "resource-metering-recorder";
+
+pub const RESOURCE_METERING_SINGLE_TARGET_THREAD: &str =
```

Comment:
Better to limit the size of thread name <= 16 bytes, referring to https://man7.org/linux/man-pages/man5/proc_pid_comm.5.html .